Sources/NIOPerformanceTester/SchedulingBenchmark.swift

```diff
@@ -18,36 +18,44 @@ import NIOPosix
 
 final class SchedulingBenchmark: Benchmark {
     private var group: MultiThreadedEventLoopGroup!
-    private var loop: EventLoop!
+    private var loops: EventLoopIterator!
     private let numTasks: Int
 
     init(numTasks: Int) {
         self.numTasks = numTasks
     }
 
-    func setUp() throws {
-        group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        loop = group.next()
-
-        // We are preheating the EL to avoid growing the `ScheduledTask` `PriorityQueue`
-        // during the actual test
-        try! self.loop.submit {
-            var counter: Int = 0
-            for _ in 0..<self.numTasks {
-                self.loop.scheduleTask(in: .nanoseconds(0)) {
-                    counter &+= 1
+    func setUp(runs: Int) throws {
```

Comment:
wait, hang on, it already tied it to exactly one `EventLoop` `self.loop`

Reply:
This whole test is completely busted. We need to either fix it (wait for the scheduled tasks) or just delete this perf test, currently it doesn't add value.

Reply:
CC @FranzBusch

Reply:
The test used to be completely busted with broken prewarming (see: https://github.com/apple/swift-nio/pull/2645#issuecomment-1942203868).

@weissi you assert:
> @gmilos instead of introducing the runs everywhere, can we not just use an ELG with just 1 thread? Then it should be fine, right?

That's not true, because the test schedules certain number (100k) tasks, but never executes them. And because the test gets run number of times over, tasks scheduled in the previous runs are still on the ELG (1 thread or not). The prewarming has to account for this.

Reply:
@weissi where did you get these test results? https://github.com/apple/swift-nio/pull/2650#discussion_r1489288143

It looks like it's still the broken (3rd Jan re-copied) test results, right?

Reply:
I'd strongly suggest to delete this test. If we don't delete this test, then the only thing we can/should do is to spawn 1 EventLoopThread in each test's `setUp`. Relying on round robin and messing with existing groups (by enqueuing 100k tasks that will never run) is not a good idea, especially if there's a possibility that other perf tests are running on the same loop.

But again: I'd say we should delete the test, I don't think it adds value.

Reply:
Fine by me.

Reply:
@FranzBusch do you have opinions (seems like you added it originally in #2009).

Reply:
Fine by me as well.